### DISCUSSION TITLE
Cant connect U print SE plus to the cartridge writer

### DISCUSSION THREAD

**Discussion by pranavbala394** [mayrthom/CartridgeWriter_uPrint#11] [Q&A]
I have been trying for the last couple of hours to try and connect the printer to the cartridge writer, but for some reason, it just won't read. I have a screenshot of the error code it gives me. I was able to connect it to putty and read it off of there, but I can't get it to work for the cartridge writer...am I doing something wrong. I should have the right cables too, made sure that it was a null modem cable...links to what I bought are below as well.

modem cable link: https://www.amazon.com/dp/B00QM8ZPZ4?psc=1&ref=ppx_yo2ov_dt_b_product_details
serial to USB adapter link: https://www.amazon.com/dp/B000A1KD1K?psc=1&ref=ppx_yo2ov_dt_b_product_details

With the proper cables and being able to read it in putty, I am really not sure what I did wrong. Also made sure that it was the latest .exe file.

![Screenshot 2022-06-20 222819](https://user-images.githubusercontent.com/107900526/174703537-012aaec0-d936-404d-ad1e-e2262fccb0f0.jpg)

**Comment by mayrthom** [comment]
If you can read with putty, you have bought the right cable. The error message says, that something is blocking the communication with the COM-Port. Did you opened another connection to the Port in another Program(e.g. Putty) at the same time?
Have you already tried to run the program as administrator?

I had this issue only when I had opened the Port of the Printer in another Program.

Best Regards

Thomas

**Reply by pranavbala394** [reply]
I am the administrator on my computer, so I dont think that is the issue. I
did have putty open at the same time...do you think that this could have
caused the error? I will try again tomorrow and without leaving putty open
and see what it says

I also have a uprint se plus, which might be another issue because I saw
that the program wasn't tested on the printer, but there is still
compatibility.

**Reply by pranavbala394** [reply]
sorry realized i had typed in the wrong spot so i copied again here

**Comment by pranavbala394** [comment]
I am the administrator on my computer, so I dont think that is the issue. I
did have putty open at the same time...do you think that this could have
caused the error? I will try again tomorrow and without leaving putty open
and see what it says

**Reply by saper-2** [reply]
Use ProcessExplorer and search what opening you COM6, it might be some slicer (e.g. cura have by default enabled plugin for connecting to printer which try to connect to any usb-rs converter that it'll detect 😑) or printer control software and it try to open the com port....

**Reply by mayrthom** [reply]
That's a good idea to check if another program has opened the COM Port.

**Comment by pranavbala394** [comment]
I also have a uprint se plus, which might be another issue because I saw
that the program wasn't tested on the printer, but there is still
compatibility.

**Reply by mayrthom** [reply]
It should work fine with the uPrint SE Plus as far as I know.
That's definitely not the source of this problem.

**Comment by sunjwd** [comment]
FYI I have just (I know, it is June of 2023 and who else is still using their Uprint SE Plus) successfully rewrote one of my material EEPROMs to 100% on our Uprint SE Plus.  I'm using version 1.4.1 so I know it works for reals.  I was able to do this with 3 (red) material EEPROMs.  Next step is to try with a 3rd party reel of ABS - on order from amazon.ca.  I did run into the OP's problem.  I ran cartridge writer on Windows 11.  I had to make sure PuTTY wasn't open at the same time as CartridgeWriter.  I also had to unplug my usb-serial-null-modem, close CartridgeWriter, re-insert usb-serial-null-modem cable, restart CartridgeWriter.  Otherwise it was in some state where it stopped with the "wrong device" message.

**Reply by sunjwd** [reply]
PS, version 1.4.1 did not have the comma vs period issue.  Just generate and write.  It worked for me, running on Windows 11 with Uprint SE Plus.

**Reply by sunjwd** [reply]
PPS: I found that cartridgewrite.uprint only works when you run it (via double click) from the folder where it is located.  Using Windows 11 search to run it didn't work for some reason.
